app/models/factory_log.rb

```diff
@@ -2,43 +2,67 @@ class FactoryLog < ActiveRecord::Base
   #TODO: fix algorithm
   #TODO: fix name devide to name and traits
   #TODO: fix visualize association relation and selectable counting or not.
+
+  # Return Factory's log ranking about total time it took.
+  # They are identified factory_name + trait_name.
   def self.time_ranking
     ranking = []
     FactoryLog.all.each do |log|
       if index = ranking.index{|cd| cd[:name] == (log.name + log.traits)}
-        ranking[index][:total_time] += log[:time]
-        ranking[index][:count] += 1
-        ranking[index][:average_time] = ranking[index][:total_time]/ranking[index][:count]
+        hash_data = ranking[index]
+        hash_data[:total_time] += log[:time]
+        hash_data[:count] += 1
+        hash_data[:average_time] = hash_data[:total_time] / hash_data[:count]
       else
-       ranking.push({name: (log.name + log.traits), total_time:log.time, count: 1, average_time: log.time})
+       ranking.push( {
+                      name: (log.name + log.traits),
+                      total_time:log.time,
+                      count: 1,
+                      average_time: log.time
+                      } )
```

Comment:
Indent the right brace the same as the first position after the preceding left parenthesis.<br>Space inside parentheses detected.